tests/phpunit/Generator/UtilsTest.php

```diff
@@ -0,0 +1,232 @@
+            $defaultQuery,
+            '#sourceContainerUrl',
+            'sourceContainerUrl6336ebdee0b80',
+            new SnowflakeQuote(),
+            '[',
+            ']',
+            <<<SQL
+                COPY INTO "stageSchemaName6336e8dda7606"."stageTableName6336e8dda7607"
+                FROM [sourceContainerUrl]
+            SQL,
+        ];
+        yield 'test generated id at the beginning' => [
+            $dedupQuerySnowflake,
+            '^stageDeduplicationTableName',
+            '__temp_DEDUP_',
+            new SnowflakeQuote(),
+            '{{ ',
+            ' }}',
+            <<<SQL
+                COPY INTO "stageSchemaName6336e8dda7606".{{ id(stageDeduplicationTableName) }}
+                FROM 'sourceContainerUrl6336ebdee0b80'
+            SQL,
+        ];
+        yield 'test generated id at the end - synapse' => [
+            $dedupQuerySynapse,
+            '$destDeduplicationTableName',
+            '_tmp',
+            new SynapseQuote(),
+            '{{ ',
+            ' }}',
+            <<<SQL
+                CREATE TABLE
+                    [destSchemaName6336e8dda7606].{{ id(destDeduplicationTableName) }}
+                WITH (DISTRIBUTION=ROUND_ROBIN,CLUSTERED COLUMNSTORE INDEX)
+            SQL,
+        ];
+        yield 'test generated id at the end - not found - synapse' => [
+            $dedupQueryWithRenameSynapse,
+            '$destDeduplicationTableName',
+            '_tmp',
+            new SynapseQuote(),
+            '{{ ',
+            ' }}',
+            <<<SQL
+                CREATE TABLE
+                    [destSchemaName6336e8dda7606].[destTableName634fca7a22355200942535tmp634fca7a3eb402_tmp_rename]
+                WITH (DISTRIBUTION=ROUND_ROBIN,CLUSTERED COLUMNSTORE INDEX)
+            SQL,
+        ];
+        yield 'test generated id at the end with rename - synapse' => [
+            $dedupQueryWithRenameSynapse,
+            '$destDeduplicationTableName',
+            '_tmp_rename',
+            new SynapseQuote(),
+            '{{ ',
+            ' }}',
+            <<<SQL
+                CREATE TABLE
+                    [destSchemaName6336e8dda7606].{{ id(destDeduplicationTableName) }}
+                WITH (DISTRIBUTION=ROUND_ROBIN,CLUSTERED COLUMNSTORE INDEX)
+            SQL,
+        ];
+        yield 'test prefixed value produces Twig syntax - synapse' => [
+            $defaultQueryWithSecretSynapse,
+            // '#' means value + Twig syntax
+            '#\'?\' ~ sourceSasSecret',
+            // prefixed by '?'
+            '?sourceSasToken634ff46baec58062943542',
+            new SynapseQuote(),
+            '{{ ',
+            ' }}',
+            <<<SQL
+                COPY INTO [stageSchemaName634ff46baec71046847136].[__temp_stageTableName634ff46baec72821993597]
+                FROM 'sourceFile1634ff46baec6c521446965'
+                WITH ( CREDENTIAL=(IDENTITY='Shared Access Signature', SECRET={{ '?' ~ sourceSasSecret }}) )
```

Comment:
nejsu si jistý jestli ten name parametr bude fungovat takto, doctrine na to má `:param` s dvojtečků. A řekl bych, že SQL server používá `@param` syntax. Možná bych použil nepojmenované parametry :thinking:  ale vlastně nevím.

Reply:
SECRET jsme vyhodili, pac se ted vsude pouziva Managed identity.
Ale aby se secret nevypisoval do query, tak bude vhodne pouzit funkci, napr. `secret(...)`.

Reply:
mělo by tam imho pak byt `SECRET=?` a udělat parameter bind v SQL, tak abysme ten secret nelogovali, případně ho v eventě replacnůt za hvězdičky